lib/private/AppConfig.php

```diff
@@ -30,413 +33,1041 @@
+		return $cache[$app][$key] ?? $default;
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param int $default default value
+	 * @param string $lazy name of the lazy group
+	 *
+	 * @return int stored config value or $default if not set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueString()
+	 * @see self::getValueFloat()
+	 * @see self::getValueBool()
+	 * @see self::getValueArray()
+	 */
+	public function getValueInt(string $app, string $key, int $default = 0, bool $lazy = false): int {
+		return (int)$this->getValueString($app, $key, (string)$default, $lazy);
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param float $default default value (optional)
+	 * @param string $lazy name of the lazy group (optional)
+	 *
+	 * @return float stored config value or $default if not set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueString()
+	 * @see self::getValueInt()
+	 * @see self::getValueBool()
+	 * @see self::getValueArray()
+	 */
+	public function getValueFloat(string $app, string $key, float $default = 0, bool $lazy = false): float {
+		return (float)$this->getValueString($app, $key, (string)$default, $lazy);
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param bool $default default value (optional)
+	 * @param string $lazy name of the lazy group (optional)
+	 *
+	 * @return bool stored config value or $default if not set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueString()
+	 * @see self::getValueInt()
+	 * @see self::getValueFloat()
+	 * @see self::getValueArray()
+	 */
+	public function getValueBool(string $app, string $key, bool $default = false, bool $lazy = false): bool {
+		return in_array(
+			$this->getValueString($app, $key, $default ? 'true' : 'false'), ['1', 'true', 'yes', 'on'], $lazy
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param array $default default value (optional)
+	 * @param string $lazy name of the lazy group (optional)
+	 *
+	 * @return array stored config value or $default if not set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy grouping
+	 * @see self::getValueString()
+	 * @see self::getValueInt()
+	 * @see self::getValueFloat()
+	 * @see self::getValueBool()
+	 */
+	public function getValueArray(string $app, string $key, array $default = [], bool $lazy = false): array {
+		try {
+			$defaultJson = json_encode($default, JSON_THROW_ON_ERROR);
+			$value = json_decode(
+				$this->getValueString($app, $key, $defaultJson, $lazy), true, JSON_THROW_ON_ERROR
+			);
+
+			return (is_array($value)) ? $value : [$value];
```

Comment:
Clean code in new methods is better then allowing bad patterns from the start:
```suggestion
			return is_array($value) ? $value : [];
```